### PR TITLE
Add main window tests for reflectometry GUI

### LINKED ISSUE #26498: Add main window tests for reflectometry GUI
The ISIS Reflectometry GUI currently does not have any tests for the MainWindowPresenter. These should be added and checked against the old tests in `ReflMainWindowPresenterTest.h` from v4.0.0.

### PR DESCRIPTION
**Pull request by gemmaguest**
This PR adds missing unit tests for the MainWindowPresenter in the ISIS Reflectometry GUI.
- An interface and mock have been added for the `BatchPresenterFactory` so that we can test creation of (and calls to) the child `BatchPresenter`s.
- The `MockMainWindowView` and `MockMainWindowPresenter` have been moved to their own files.
- This PR also includes minor fix where there was a typo causing an incorrect notification function to be called. Fortunately this did not cause a bug in the real interface because currently the two notification functions both end up calling the same underlying function to update the widget state.

**To test:**

This should be tested at ISIS.

- Code review, and ensure unit tests pass
- Open the ISIS Reflectometry interface and check that Batches are created ok: click the menu Batch->New and check a new Batch is added. Try removing a batch with the X on the tab (note that there is a known bug where the batch numbers may not be unique).
- Check that autoprocessing still works ok:
  - The Autoprocess button should be enabled by default and the Pause button disabed on all Batch tabs.
  - Enter instrument INTER and investigation id `1120015` and click Autoprocess (log in to ICat if prompted).
  - The Autoprocess buttons on all tabs should be disabled and the Pause button enabled on the batch that is processing.
  - Click Pause. It should revert to the original state.

Fixes #26498 	

*This does not require release notes* because **it concerns unit testing only**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by dtasev — CHANGES_REQUESTED** [review]
Just a small change on one of the includes.

Overall really good - I really like the style of moving expectations/assertions into functions, it makes the tests read like a little story! It's something I'll incorporate into my future unit tests too

File: qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/MainWindowView.h [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #ifndef MANTID_ISISREFLECTOMETRY_MAINWINDOWVIEW_H
 #define MANTID_ISISREFLECTOMETRY_MAINWINDOWVIEW_H
 
+#include "GUI/Batch/IBatchPresenterFactory.h"
```
[the remark this answers is not in the document]
Comment (gemmaguest) [reply qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/MainWindowView.h]: Ah, that is weird - thanks for that, would have taken me a while to work that one out!

**Comment by DTasev** [comment]
Some merge conflicts have also appeared

**Comment by gemmaguest** [comment]
@DTasev The conflicts are fixed and it's finally compiling - the move constructor and assignment operator also needed moving to the cpp file

**Review by dtasev — APPROVED** [review]
Tests finally compile and no merge conflicts! LGTM

**Pull request merged**